clang/lib/Driver/Driver.cpp

```diff
@@ -5524,9 +5524,11 @@ InputInfo Driver::BuildJobsForActionNoCache(
   else {
     std::string OffloadingPrefix;
     // When generating binaries with -fsycl-link-target or -fsycl-link, the
-    // output file prefix is the triple arch only.
+    // output file prefix is the triple arch only.  Do not add the arch when
+    // compiling for FPGA with -fsycl-link.
     if (Args.getLastArg(options::OPT_fsycl_link_targets_EQ) ||
```

Comment:
How is OPT_fsycl_link_targets_EQ different from OPT_fsycl_link_EQ? Could the former also have OPT_fintelfpga set?

Reply:
The use of `-fsycl-link-targets` is part of an earlier methodology for performing AOT compilation.  Where AOT was actually done by hand and not automatic in the driver.  We add the arch naming to make sure that the output names are unique when using multiple targets.  `-fsycl-link-targets` is not meant to be used with `-fintelfpga`